### ISSUE TITLE
Update lint extras for py39

### ISSUE THREAD

**Issue by Hakuyume** [pfnet/pysen#22]
Since PFN uses py39 widely, `pysen[lint]` should work with py39.
Current lint extras tries to install older black and it does not work with py39.

(optional) Can we drop py36 and py37 support?

**Comment by sergeant-wizard** [comment]
The error in question can be reproduced by running the following at the root of pysen repo:

```
docker run --workdir /ws -v $(pwd):/ws python:3.9-slim-buster bash -c "pip install pysen[lint] && pysen run lint"
```

The issue is reported here as well: https://github.com/python/typed_ast/issues/169

**Comment by bonprosoft** [comment]
Thank you! :+1: to upgrade `black` to work with py39.

**Comment by sergeant-wizard** [comment]
> The error in question can be reproduced by running the following at the root of pysen repo:
> 
> ```
> docker run --workdir /ws -v $(pwd):/ws python:3.9-slim-buster bash -c "pip install pysen[lint] && pysen run lint"
> ```
> 
> The issue is reported here as well: [python/typed_ast#169](https://github.com/python/typed_ast/issues/169)

I confirmed that the error from black in question is resolved in `0.10.3`
